tka/sig_test.go

```diff
@@ -30,11 +32,54 @@ func TestSigDirect(t *testing.T) {
 		t.Errorf("sigHash changed after signing: %x != %x", sig.SigHash(), sigHash)
 	}
 
-	if err := sig.verifySignature(key); err != nil {
+	if err := sig.verifySignature(node.Public(), k); err != nil {
 		t.Fatalf("verifySignature() failed: %v", err)
 	}
 }
 
+func TestSigNested(t *testing.T) {
```

Comment:
Suggestion for additional tests:
1. Deeply nested signature (100+)
2. First signature fails
3. Last signature fails
4. An intermediate signature fails
5. The last signature is valid but not actually on the approver list

Reply:
Done.